tests/integration/records/test_records_api.py

```diff
@@ -180,3 +183,157 @@ def test_not_count_citations_from_other_collections(isolated_app):
     TestRecordMetadata.create_from_kwargs(json=ref)
 
     assert record_1.get_citations_count() == 1
+
+
+def test_get_modified_references_when_prev_version_has_zero(app):
+    json = {
```

Comment:
Maybe I'm missing something (like the proper signals are not triggered when using factories??), but to me it looks like you could have used the factories.

json_data = []
json_data['references'] = [...]
factory = TestRecordMetadata.create_from_kwargs(json=json_data)
# factory.inspire_record  # if you need to access the InspireRecord instance